kubernetes-helm: 3.8.1 -> 3.8.2

techknowlogick opened the pull request:
###### Description of changes
https://github.com/helm/helm/releases/tag/v3.8.2

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

techknowlogick committed:
kubernetes-helm: 3.8.1 -> 3.8.2

saschagrunert commented on pkgs/applications/networking/cluster/helm/default.nix:
```diff
@@ -2,16 +2,16 @@
   pname = "helm";
-  version = "3.8.1";
+  version = "3.8.2";
   gitCommit = "5cb9af4b1b271d11d7a97a71df3ac337dd94ad37";
```
Let's change this one as well.

techknowlogick replied:
Thanks for catching that :) I've just updated.

Artturin commented on pkgs/applications/networking/cluster/helm/default.nix:
```diff
@@ -2,16 +2,16 @@
-  vendorSha256 = "sha256-ffbp7J8XDxo/s79pjoiDVbft0pr/lJpuJuKiMpQwkT0=";
+  vendorSha256 = "sha256-FLEydmR+UEZ80VYLxBU1ZdwpdLgTjUpqiMItnt9UuLY=";
 
   doCheck = false;
```
Why? Please add a comment and a passthru.tests.version test

Chili-Man replied:
that change was already pre-existing

techknowlogick replied:
Yes, that's pre-existing. I'd much rather address that in a different PR though (unless @superherointj beats me to it as they've been sending many PRs recently enabling tests in nix go packages)

superherointj replied:
You should have added here. I only add tests in a separate PR when I don't have any other change in mind. A single commit per package.

techknowlogick replied:
@superherointj no worries. I was able to cleanly apply that PR to this one.

superherointj commented:
@saschagrunert Notice:
![image](https://user-images.githubusercontent.com/5861043/164478515-6a68908f-a290-4e33-8e9d-a49b8a443c76.png)
Which platform have you built? Maybe the extra patch should be done for Linux only.

techknowlogick commented:
I've pushed an update that hopefully deals with the failures on linux, although if someone w/ a linux machine could review this PR, I'd be very appreciative.

bryanasdev000 reviewed (approved):
LGTM:

```
$ nixpkgs-review pr 168515 

...

2 packages built:
kubernetes-helm telepresence2
```

``` 
$ helm version
version.BuildInfo{Version:"v3.8.2", GitCommit:"6e3701edea09e5d55a8ca2aae03a68917630e91b", GitTreeState:"", GoVersion:"go1.17.9"}
```

superherointj commented:
> I've pushed an update that hopefully deals with the failures on linux, although if someone w/ a linux machine could review this PR, I'd be very appreciative.

@techknowlogick Can you correct the 3 issues I pointed out?
1) If it does not error for Darwin, should skip for Linux only.
2) Rename package to "kubernetes-helm" to match top-level.
3) Add `mainProgram = "helm";`

techknowlogick commented:
@superherointj thanks :) I've made those changes.

superherointj commented on pkgs/applications/networking/cluster/helm/default.nix:
```diff
@@ -23,6 +22,19 @@ buildGoModule rec {
+    # skipping version tests because they require dot git directory
+    substituteInPlace cmd/helm/version_test.go \
+      --replace "TestVersion" "SkipVersion"
+  '' + lib.optionalString (!stdenv.isDarwin) ''
```
```suggestion
  '' + lib.optionalString stdenv.isLinux ''
```

superherointj commented:
> @superherointj thanks :) I've made those changes.

Did you add `mainProgram = "helm";` to meta?

IMO, `stdenv.isLinux` is more precise. Shorter also.

techknowlogick commented:
> Did you add mainProgram = "helm"; to meta?

whoops, you're right. Added it to the wrong place. Thanks again for all your patience.

>  IMO, stdenv.isLinux is more precise. Shorter also.

updated

superherointj reviewed (approved):
Built successfully `linux-x86_64`

superherointj reviewed (approved):
LGTM

thiagokokada commented:
Result of `nixpkgs-review pr 168515` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>kubernetes-helm</li>
    <li>telepresence2</li>
  </ul>
</details>
